src/self_encryptor.rs

```diff
@@ -1483,6 +1715,81 @@ mod tests {
         Ok(())
     }
 
+    async fn general_truncate_test(
+        old_size: u32,
+        new_size: u32,
+        num_chunks: u32,
+    ) -> Result<(), SelfEncryptionError> {
+        let bytes_len = old_size;
+        let mut rng = new_test_rng()?;
+        let bytes = random_bytes(&mut rng, bytes_len as usize);
+        let (data_map, storage) = {
+            let storage = SimpleStorage::new();
+            let se = SelfEncryptor::new(storage, DataMap::None)
+                .expect("First encryptor construction shouldn't fail.");
+            se.write(&bytes, 0)
+                .await
+                .expect("Writing to encryptor shouldn't fail.");
+            check_file_size(&se, bytes_len as u64).await;
+            se.close().await?
+        };
+
+        let (data_map2, storage) = {
+            // Start with an existing data_map.
+            let se = SelfEncryptor::new(storage, data_map)
+                .expect("Second encryptor construction shouldn't fail.");
+            se.truncate(new_size as u64)
+                .await
+                .expect("Truncating encryptor shouldn't fail.");
+            se.close().await?
+        };
+
+        assert_eq!(data_map2.len(), new_size as u64);
+        match data_map2 {
+            DataMap::Chunks(ref chunks) => {
+                assert_eq!(chunks.len(), num_chunks as usize);
+                for chunk_detail in chunks.iter() {
+                    assert!(storage.has_chunk(&chunk_detail.hash));
+                }
+            }
+            _ => panic!("data_map should be DataMap::Chunks"),
+        }
+
+        let min_size = cmp::min(old_size, new_size);
+        let se = SelfEncryptor::new(storage, data_map2)
+            .expect("Third encryptor construction shouldn't fail.");
+        let fetched = se
+            .read(0, min_size as u64)
+            .await
+            .expect("Reading from encryptor shouldn't fail.");
+        assert_eq!(&fetched[..], &bytes[..(min_size) as usize]);
+        Ok(())
+    }
+    #[tokio::test]
+    async fn truncate_three_to_four_chunk() -> Result<(), SelfEncryptionError> {
```

Comment:
better have a comment explain the general setup and expected result of the test.
same comment to the below tests.